### PR TITLE
feat/Contributor and changelog schemas

### LINKED ISSUE #9: chore/authors, contibutors & changelog schemas
Current version of the schema idea: (see edits for previous versions)
Screenshot
[Source file](https://github.com/user-attachments/files/17792663/ToPWR.contributors.changelogs.schema_2024-11-17T22_30_36.726Z.ddb.json)

### LINKED ISSUE #42: feat/add adonis models for authors/contributors & changelogs

### LINKED ISSUE #43: feat/add db migrations for authors/contributors & changelogs

### PR DESCRIPTION
**Pull request by mini-bomba**
This PR implements the contributor and changelog schemas discussed in #9.
diagram

A few changes were made since the last update in #9:
- `social_link_types` table was converted into an enum shared with the `departments` table
- all foreign key fields now have the `_id` suffix
- `photo` column on the `contributors` table was renamed to `photo_key` to suggest that image keys should be stored there instead of URLs
- `screenshot` columns on the `*_screenshots` tables were renamed to `image_key` - same reasoning as above
- `change_types` table was converted into a new enum
- `version_changelog` pivot table was deleted in favour of a one-to-many relationship between `versions` and `changes`, as a many-to-many relationship does not make sense
- a `description` optional field was added to the `versions` table
- `version` column on the `versions` table was renamed to `name`

closes #9 
closes #42 
closes #43

### CODE REVIEW

File: database/migrations/1733096077721_create_contributor_social_links_table.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'tiktok',
+    'discord',
+    'twitch',
+  ]
+
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id')
+      table
+        .integer('contributor_id')
```
Comment (czaja307) [inline database/migrations/1733096077721_create_contributor_social_links_table.ts:25]: it's better to use .foreign imo
https://lucid.adonisjs.com/docs/table-builder#foreign

**Review by czaja307 — COMMENTED** [review]
soldid work, just change foreign keys in migrations to use .foreign

File: database/migrations/1733096077721_create_contributor_social_links_table.ts (hunk 1 above)
Comment (mini-bomba) [reply database/migrations/1733096077721_create_contributor_social_links_table.ts:25]: aight, will do

Comment (czaja307) [reply database/migrations/1733096077721_create_contributor_social_links_table.ts]: i don't think you need to split it into `.integer` and `.foreign` parts
just `.foreign` seems to be doing just fine

**Review by czaja307 — COMMENTED** [review]
i don't think you need to split it into .integer and .foreign parts
just .foreign seems to be doing just fine

**Comment by mini-bomba** [comment]
image
misleading documentation smh

**Review by czaja307 — DISMISSED** [review]

**Review by czaja307 — APPROVED** [review]

**Pull request merged**